ignite/handlers/checkpoint.py

```diff
@@ -295,14 +312,21 @@ def __call__(self, engine: Engine) -> None:
                 for k in checkpoint:
                     name = k
                 checkpoint = checkpoint[name]
-            filename = "{}{}_{}{}".format(self._fname_prefix, name, suffix, self._ext)
+
+            filename_pattern_dict["name"] = name
+            filename_prefix = self.filename_prefix
+            if self.filename_pattern is None:
```

Comment:
Here we have 2 paths : in a case, the dict is useless, and for the other case, suffix is useless. Maybe we should refactor following the code I provided ? (See other comment). 

Btw, thank you again, we are close to a perfect score 😉

Reply:
Will update accordingly @sdesrozis let's make this perfect 💪 .